api/src/main.rs

```diff
@@ -10,14 +12,53 @@ struct Args {
 
 #[derive(Subcommand, Clone)]
 enum Command {
-    Run { cmdline: Vec<String> },
+    Run {
+        cmdline: Vec<String>,
+
+        #[arg(short = 'e', long = "env")]
+        environment: Vec<String>,
+    },
+}
+
+// Parse a list of arguments given to the -e flag, and returns a map from key to value:
+// - If the argument has the form KEY=VALUE, then (KEY, VALUE) is used
+// - If the argument has the form KEY and KEY exists in the current process' environment, that
+//   value is used
+// - If the arguments has the form KEY and KEY does not exist in the process' environment, the
+//   argument is ignored
+//
+// This is the same behaviour used by the `docker run` command.
+fn parse_environment(env: &[String]) -> anyhow::Result<HashMap<String, String>> {
```

Comment:
Can you put some unit tests on this that validate the above comments? It all looks straightforward enough, but worth testing I think. It's possible that some of the transient environment variable setting bits that would be nice to use in tests will be a faff though

Reply:
Done. Setting environment variables at all is a bit of a minefield in a multithreaded program (which tests are). I've fixed that by passing the function to get environment variables as an argument and "mocking" it in the tests